### DISCUSSION TITLE
v1.1.0 focus

### DISCUSSION THREAD

**Discussion by Shawoll** [jreleaser/jreleaser#841] [General]
Hi everybody,
I would like to ask you, if I can somehow help with the version `v1.1.0`. There are interesting features, I would like to incorporate into my project. Especially `scp` and `sftp` upload. I've found, at the time of writing this, 4 opened issues for version v1.1.0. Unfortunately, I don't have a Apple Silicon (M1). My other choices are one of the #809 , #806 , #664 issues. Which one of these is doable, without any external blockers? 

Also, I am using `jreleaser` with `Gradle Kotlin DSL`, but all documentation I've found so far, supports `Gradle Groovy DSL`. If possible, I would like to contribute into that as well. 
If there's anything I can help you with, please let me know.

Although, I have some experience with sw development, I never contributed to any OpenSource project before, but I would love to begin somewhere, and IMHO, JReleaser has a huge potential. I love the whole idea behind it. 
If I posted something not correct into this discussion area, please let me know and I'll more than happy to rewrite/edit this post. 
Thanks in advance.

**Comment by aalmiray** [comment]
Hi @Shawoll, thank you for taking an interest in contributing to JReleaser.

Issues #809 and #806 require new capabilities that have yet to be added. These issues would benefit from adding a new section, `renderer`, that would allow customizing how commit messages should be rendered. Rendering rules may be applied based on labels, or the title and/or body content.

#664 is partly done as the project provides executables for MacOs and Linux. Windows executables are currently blocked by a pair of issues when running inside GH actions.

#840 is a good starting issue as the missing code is located at `org.jreleaser.assemblersJpackageAssemblerProcessor` by line 227.

You could also provide an SK translation for the CLI resources if you like 😄 

Finally, adding Kotlin DSL snippets is a big undertaking and must be performed at https://github.com/jreleaser/jreleaser.github.io
I decided against adding those snippets for a couple of reasons:
1. I don't use the Kotlin DSL thus I have no experience with it
2. IntelliJ provides code migration from Groovy to Kotlin
3. Kotlin DSL is only usable with IntelliJ
4. Let IntelliJ do the work

Even if Kotlin DSL were to be added as a 6th variant it would have to be added to ALL configuration snippets AND be maintained going forward.

**Reply by Shawoll** [reply]
Hi, thanks for the reply! Regarding Kotlin DSL, you're right. 
About the translation, sure. I'll be happy to contribute that with that part and also, the issue you've mentioned seems fine for the starter. 
Thank you, I'll keep you guys updated with the progress.
